packages/components/src/components/input-field/input.tsx

```diff
@@ -128,7 +128,7 @@ const Input = ({
                 data-value={data_value}
                 disabled={!!is_disabled}
                 id={id}
-                maxLength={fractional_digits ? max_length + fractional_digits + 1 : max_length}
+                maxLength={fractional_digits && max_length ? max_length + fractional_digits + 1 : max_length}
```

Comment:
@kate-deriv this will break if max_length is 0

Reply:
I think that would fix:
`maxLength={fractional_digits && (max_length ||  max_length === 0) ? max_length + fractional_digits + 1 : max_length}`